drivers/SmartThings/zwave-switch/src/multi-metering-switch/init.lua

```diff
@@ -0,0 +1,140 @@
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+local st_device = require "st.device"
+local capabilities = require "st.capabilities"
+--- @type st.zwave.CommandClass
+local cc = require "st.zwave.CommandClass"
+--- @type st.zwave.CommandClass.Meter
+local Meter = (require "st.zwave.CommandClass.Meter")({version = 3})
+--- @type st.zwave.CommandClass.Basic
+local Basic = (require "st.zwave.CommandClass.Basic")({ version = 1, strict = true })
+--- @type st.zwave.CommandClass.SwitchBinary
+local SwitchBinary = (require "st.zwave.CommandClass.SwitchBinary")({version = 2, strict = true })
+
+local PARENT_ENDPOINT = 1
+
+local MULTI_METERING_SWITCH_FINGERPRINTS = {
+  {mfr = 0x0086, prod = 0x0003, model = 0x0084}, -- Aeotec Nano Switch 1
+  {mfr = 0x0086, prod = 0x0103, model = 0x0084}, -- Aeotec Nano Switch 1
+  {mfr = 0x0086, prod = 0x0203, model = 0x0084}, -- AU Aeotec Nano Switch 1
+  {mfr = 0x027A, prod = 0xA000, model = 0xA004}, -- Zooz ZEN Power Strip 1
+  {mfr = 0x015F, prod = 0x3102, model = 0x0201}, -- WYFY Touch 1-button Switch
+  {mfr = 0x015F, prod = 0x3102, model = 0x0202}, -- WYFY Touch 2-button Switch
+  {mfr = 0x015F, prod = 0x3102, model = 0x0204}, -- WYFY Touch 4-button Switch
+  {mfr = 0x015F, prod = 0x3111, model = 0x5102}, -- WYFY Touch 1-button Switch
+  {mfr = 0x015F, prod = 0x3121, model = 0x5102}, -- WYFY Touch 2-button Switch
+  {mfr = 0x015F, prod = 0x3141, model = 0x5102}, -- WYFY Touch 4-button Switch
+}
+
+local function can_handle_multi_metering_switch(opts, driver, device, ...)
+  for _, fingerprint in ipairs(MULTI_METERING_SWITCH_FINGERPRINTS) do
+    if device:id_match(fingerprint.mfr, fingerprint.prod, fingerprint.model) then
+      return true
+    end
+  end
+  return false
+end
+
+local function get_profile(device)
+  if device:get_manufacturer() == 0x015F then -- only WYFY's have switch-binary
+    return "switch-binary"
+  else
+    return "metering-switch"
+  end
+end
+
+local function is_zooz_power_strip(device)
+  if device:get_manufacturer() == 0x027A and device:get_model() == 0xA004 then
+    return true
+  end
+end
+
+local function create_child_device(driver, device, children_amount)
+  for i = 2, children_amount+1, 1 do
+    local device_name_without_number = string.sub(driver.label, 0,-2)
+    local name = string.format("%s%d", device_name_without_number, i)
+    local metadata = {
+      type = "EDGE_CHILD",
+      label = name,
+      profile = get_profile(device),
+      parent_device_id = device.id,
+      parent_assigned_child_key = string.format("%02X", i),
+      vendor_provided_label = name,
+    }
+    driver:try_create_device(metadata)
+  end
+end
+
+local function device_added(driver, device, event)
+  if device.network_type == st_device.NETWORK_TYPE_ZWAVE then
+    local children_amount
+    if is_zooz_power_strip(device) then
+      children_amount = 4;
+    else
+      children_amount = device.zwave_endpoints
```

Comment:
Since `device.zwave_endpoints` is a table, shouldn't You get the length of it here ? (excluding main endpoint)

Reply:
fixed